cartography/intel/aws/ec2/launch_templates.py

```diff
@@ -22,7 +22,19 @@ def get_launch_templates(boto3_session: boto3.session.Session, region: str) -> l
     paginator = client.get_paginator('describe_launch_templates')
     templates: list[dict[str, Any]] = []
     for page in paginator.paginate():
-        templates.extend(page['LaunchTemplates'])
+        paginated_templates = page['LaunchTemplates']
+        for template in paginated_templates:
+            try:
```

Comment:
2 options:

Option 1. **Simplest**: I think we should remove the try+catch here. The previous version of the code (https://github.com/lyft/cartography/blob/21bd5a379bb5a992d6e540c4b70c9228fa2f15c4/cartography/intel/aws/ec2/launch_templates.py#L24-L29) did not have the try+catch and did not have the race condition (that I introduced :x).

The only situation where we should be catching the base `Exception` is in "top level" orchestrator code. There are too many things that can go wrong 

Option 2. **more time-consuming**: catch the `botocore.exceptions.ClientError` and check specifically for that `InvalidLaunchTemplateId.NotFound` error code. I started on this briefly but got frustrated when I needed to add in boto3-stubs as a dependency to properly unit test it. Boto3 does all kinds of dynamic metaprogramming stuff that make it hard to unit test.

Given that I don't recall Ryan's old code throwing the race condition, I think removing the try+catch is best. We probably don't have that race condition anymore because the call to retrieve template versions happens closer to the call to retrieve templates.

Reply:
Like we discussed, we need the `Exception` in cases where the environment is so dynamic that it can create and destroy launch templates in milliseconds.
I can catch only the `botocore.exceptions.ClientError`, but that is too specific to https://github.com/lyft/cartography/issues/1321, I would prefer to keep it simple and manage all sort of scenarios skipping the launch templates not found.

Reply:
Ah, in our convo I mentioned that I am ok with the try+except, but we shouldn't catch the base `Exception`. There are other things that can fail in the code block and we'd run the risk of burying other things.